### DISCUSSION TITLE
Sync List only including first entry/line

### DISCUSSION THREAD

**Discussion by Weissrolf** [abraunegg/onedrive#2869] [Q&A] [answered]
Hello.

Did the format of the sync list change with v2.5.2? Onedrive only syncs the first include entry/line, but does not seem to sync the second one anymore.

```
Minecraft*
RPG/FoundryVTT*
```

Thanks and regards.

**Comment by abraunegg** [comment, marked as answer]
> Did the format of the sync list change with v2.5.2?

Not with v2.5.2. The entire v2.5.x is a 100% re-write.

> Onedrive only syncs the first include entry/line, but does not seem to sync the second one

Please read the documentation on correctly formatting your 'sync_list'

Your 'entries' are *searching* for those paths *anywhere*

Correct your 'sync_list' as per the documentation, including your exclude line which is also not formatted correctly.

**Comment by Weissrolf** [comment]
Yes, I know that the entries are searching for the paths anywhere. That's intended in case I decide to move them. It's works fine for the Minecraft folder and subfolders, but not for the FoundryVTT one. The last modified subfolder date in the FoundryVTT is from last night before I updated Onedrive (I think files were not synced, though).

I tried "RPG/FoundryVTT*" (what worked before), "FoundryVTT/*" and "RPG/FoundryVTT/", all of which failed.

What *does* work is "/RPG/FoundryVTT/", but that means any other (non root/RPG/) "FoundryVTT" folder is excluded by Onedrive.

The exclude line is meant to exclude any "modules/!*" subfolder, with the ! being the first character of the subfolder name, hence "!modules/!*"

**Reply by abraunegg** [reply]
What does the application output say?

What is indicated when you enable verbose logging `--verbose` ?

**Reply by Weissrolf** [reply]
Application output only counts 11 folders to sync instead of 111 (now a bit less, because I added another exclude for thumbs). It just ignores the whole FoundryVTT folder structure. I am running a verbose --resync without root "/" now and will report back once it's finished.

`New directories to create on Microsoft OneDrive: 11
New items to upload to Microsoft OneDrive: 11`

**Reply by Weissrolf** [reply]
Yeah, FoundryVTT is not monitored unless a leading root /RPG/ is put in front of it.

```
Monitoring directory: .
Monitoring directory: ./Minecraft
Monitoring directory: ./Minecraft/crafty
Monitoring directory: ./Minecraft/crafty/crafty-4
Monitoring directory: ./Minecraft/crafty/crafty-4/backups
Monitoring directory: ./Minecraft/crafty/crafty-4/backups/5d636364-e887-4fcc-b705-6574cad1b2e2
Monitoring directory: ./Minecraft/crafty/crafty-4/backups/5d636364-e887-4fcc-b705-6574cad1b2e2/3d7eb928-c102-422d-9628-e0a40b1d0d4a
Monitoring directory: ./Minecraft/crafty/crafty-4/backups/77e508bd-0d67-40d1-a2c3-98a4e3197a10
Monitoring directory: ./Minecraft/crafty/crafty-4/backups/77e508bd-0d67-40d1-a2c3-98a4e3197a10/0cefbab3-272b-4693-bc40-cc5850f207be
Monitoring directory: ./Minecraft/crafty/crafty-4/backups/d3f1ab87-d898-4215-9758-c1110e494eb2
Monitoring directory: ./Minecraft/crafty/crafty-4/backups/d3f1ab87-d898-4215-9758-c1110e494eb2/9eac0a39-5ffb-4787-a4dd-b34d78ca5a73
Monitoring directory: ./Minecraft/crafty/crafty-4/backups/95306bea-4439-4b4b-b8e1-5a65c6ddfcce
Performing initial synchronisation to ensure consistent local state ...
```

**Comment by abraunegg** [comment]
Monitoring a directory is different to something being included / excluded by your 'sync_list' file.

That is what I would be more interested in understanding

**Reply by Weissrolf** [reply]
This works:

```
!modules/!*
!thumbs/
/Minecraft/
/RPG/FoundryVTT/
```

This doesn't work:

```
!modules/!*
!thumbs/
Minecraft/
RPG/FoundryVTT/
```
(same without the RPG/)

**Reply by abraunegg** [reply]
That was not my ask ...

What is the application doing when it details what it being included / excluded

**Reply by Weissrolf** [reply]
And there is no mentioning of "FoundryVTT" in the verbose output whatsoever for the latter version.

**Reply by Weissrolf** [reply]
The application syncs the Minecraft folder and acts like "FoundryVTT" was never included or mentioned to begin with.

**Comment by Weissrolf** [comment]
<img width="1340" alt="image" src="https://github.com/user-attachments/assets/51d87f18-2da0-4076-b56a-c3236e31c78c">

**Reply by Weissrolf** [reply]
That's it, no more output after this.

**Comment by Weissrolf** [comment]
Maybe it doesn't like the capital "VTT" characters at the end of the folder name?

**Comment by Weissrolf** [comment]
But I just spotted this single line:
<img width="558" alt="image" src="https://github.com/user-attachments/assets/a5bf3ad5-fbf4-4037-9b53-f4f53471a688">

**Comment by abraunegg** [comment]
When using 'sync_list' and `--verbose` the application details what is being included and excluded based on the sync list entries.

**Comment by Weissrolf** [comment]
`Skipping path - excluded by sync_list config: ./RPG`

**Reply by Weissrolf** [reply]
Same when I use "RPG/FoundryVTT", still "Skipping path - excluded by sync_list config: ./RPG".

**Comment by Weissrolf** [comment]
The two folders "Minecraft" and "RPG" are the only folders found in Onedrive's root on the local system. I enabled upload-only.

**Comment by abraunegg** [comment]
Please do the following:

1. Put your 'sync_list' to your default / desired state
2. Raise a new bug report
3. Generate and provide a complete verbose debug log
4. Provide a complete local folder/directory view (use the command `tree -d` to list your entire directory structure so that this can be replicated
5. Give exlicit details based on the output of 'tree' as to what you are expecting to happen
